ast/src/parsed/display.rs

```diff
@@ -1200,10 +1200,27 @@ mod tests {
                 ("let x = (1 + (|i| i + 2));", "let x = 1 + (|i| i + 2);"),
                 // Index access
                 ("(|i| i)[j];", "(|i| i)[j];"),
-                // Lambda expression
+            ];
+
+            for test_case in test_cases {
+                test_paren(&test_case);
+            }
+        }
+
+        #[test]
+        fn lambda_parentheses() {
+            let test_cases: Vec<TestCase> = vec![
+                ("1 + (|x| x)(2);", "1 + (|x| x)(2);"),
                 ("|i| (|x| x) + i;", "|i| (|x| x) + i;"),
                 ("|i| (|x| x + i);", "|i| (|x| x + i);"),
                 ("|i| |x| x + i;", "|i| (|x| x + i);"),
+                ("(|i| i) + (|x| x);", "(|i| i) + (|x| x);"),
```

Comment:
Interestingly, we would actually not need the second pair of parentheses, i.e. the following would also be ok: `(|i| i) + |x| x`, but this is good enough and maybe even better because less ambiguous.